app/javascript/src/utils/compiler.js

```diff
@@ -137,6 +138,93 @@ function extractAndInsertMixins(joinedItems) {
   return joinedItems
 }
 
+/**
+ * @param {string} source
+ * @param {number} fromIndex
+ * @returns
+ */
+function findMatchingClosingBracketIndex(source, fromIndex) {
+  let result = -1
+  let currentDepth = 1
+  for (let i = fromIndex; i < source.length; i++) {
+    const char = source[i]
+    if (char === "{") {
+      currentDepth++
+    } else if (char === "}") {
+      currentDepth--
+      if (currentDepth <= 0) {
+        result = i
+        break
+      }
+    }
+  }
+  return result
+}
+
+/**
+ * @param {string} source
+ * @returns {string}
+ */
+function evaluateConditionals(source) {
+  const conditionalStartRegex = /@if *\( *((?:.|\n)+?) *(==|!=) *((?:.|\n)+?)\) *[ \n]*\{/g
```

Comment:
Regex, man, what is even happening :D

Reply:
https://regex101.com/r/WN1vYH/1

There is a lot of fluff for the spaces ` *`.
`((?:.|\n)+?)` matches the left hand or right hand.